frontend/components/generic/SimpleTable.tsx

```diff
@@ -21,23 +21,28 @@ type TableColumn<D, K> = K extends keyof D
     : TableColumnWithDefinedTransform<D, K>
   : never;
 
+type Comparable = number | string;
+
 type TableColumnWithOptionalTransform<D, K extends keyof D> = D[K] extends React.ReactNode
   ?
       | {
           title: string;
           key: K;
+          getSortValue?: (a: D) => Comparable;
         }
       | {
           title: string;
           key: K;
           transformValue: (arg: D[K]) => React.ReactNode;
+          getSortValue?: (a: D) => Comparable;
         }
   : never;
 
 type TableColumnWithDefinedTransform<D, K extends keyof D> = {
   title: string;
   key: K;
   transformValue: (arg: D[K]) => React.ReactNode;
+  getSortValue?: (a: D) => Comparable;
```

Comment:
```suggestion
  getSortValue?: (a: D[K]) => Comparable;
```

Does it make sense if the input type is `D[K]` instead? Since the sort value should be found in `D[K]` itself.

Reply:
Also could you think of a more descriptive variable name than `a`?

Reply:
I don't believe arg / input / etc is any more descriptive than `a`, this only serves a purpose as syntax for defining the type right? Anyway I've changed it to `arg`.